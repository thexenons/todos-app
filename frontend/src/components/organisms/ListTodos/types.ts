import { Todo } from "../../../api/entities";

export interface ListTodosProps {
	todos: Todo[];
	onClickTodo?: (todo: Todo) => void;
	onBack?: () => void;
}
