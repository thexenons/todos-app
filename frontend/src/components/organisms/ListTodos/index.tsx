import { FC } from "react";

import { ListTodosProps } from "./types";

const ListTodos: FC<ListTodosProps> = ({ todos, onClickTodo, onBack }) => (
	<div>
		<h4>Todos {onBack && <button onClick={onBack}>Back</button>}</h4>
		<ul>
			{todos.map((todo) => (
				<li
					key={todo.id}
					role="presentation"
					onClick={() => onClickTodo?.(todo)}
				>
					<h5>{todo.name}</h5>
				</li>
			))}
		</ul>
	</div>
);

export default ListTodos;
