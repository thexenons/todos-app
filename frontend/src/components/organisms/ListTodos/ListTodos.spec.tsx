import { render } from "@testing-library/react";

import BaseTestComponent from "../../../tests/BaseTestComponent";
import ListTodos from ".";

describe("ListTodos", () => {
	it("Render", () => {
		const { container } = render(
			<BaseTestComponent>
				<ListTodos todos={[]} />
			</BaseTestComponent>
		);
		expect(container).toMatchSnapshot();
	});
});
