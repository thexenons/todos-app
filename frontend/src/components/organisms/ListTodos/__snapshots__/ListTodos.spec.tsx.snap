// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ListTodos > Render 1`] = `
<div>
  <div>
    <h4>
      Todos 
    </h4>
    <ul />
  </div>
</div>
`;
